Automattic/jetpack
# Add tracking analytics (Tracks) to the Backup plugin.

This PR adds Tracks tracking to the Backup plugin, and adds tracking events to the links in the Backup dashboard UI.

#### Changes proposed in this Pull Request:
- Adds `jetpack-analytics' dependency to package.json.
- Added a new `useAnalytics()` hook to initialize analytics for the Backup plugin.
- Added a tracking event for the Backup plugin admin **page-view**. (`jetpack_backup_admin_page_view`)
- Added a tracking event for the "**See backups in the cloud**" CTA button click. (`jetpack_backup_see_backups_cta_click`)
- Added a tracking event for the "**Or view your most recent restore point**" link click, (`jetpack_backup_view_recent_restore_points_click`)
- Added a tracking event for the "**Learn more**" link click. (`jetpack_backup_learn_more_click`)
- Added a tracking event for the "**See all your backups**" link click. (`jetpack_backup_see_all_backups_click`)
- Added a tracking event for the "**See your site's activity**" link click. (`jetpack_backup_see_site_activity_click`)

Screenshot:
** Red arrows indicate the links where tracking was added.
![Markup 2022-07-07 at 16 58 52](https://user-images.githubusercontent.com/11078128/177870339-c874106c-6fb2-46f9-8003-86712559c392.png)

#### Other information:


#### Jetpack product discussion


Project Thread: p1HpG7-gb4-p2

Tracking and design changes are further described in this i2 design P2: p1HpG7-gkG-p2

This PR is a continuation of: https://github.com/Automattic/jetpack/pull/24955

#### Does this pull request change what data or activity we track or use?


Yes. It adds some link click tracking.

#### Testing instructions:






- Spin up a JN site with Jetpack Beta, using Backup branch `add/backup-plugin-analytics`.
- Connect Backup and purchase Backup plan with credits (or add Backup(10GB) product using Store Admin).
- Go to https://cloud.jetpack.com/settings/:site?action=edit (replace :site with your JN site slug) and **enter your JN site server credentials**.
- Navigate to the Jetpack --> Backup page (`/wp-admin/admin.php?page=jetpack-backup`).
- Open developer tools console and enter: `localStorage.setItem('debug', 'dops:analytics');`
- Reload/refresh the Backup admin page.
- When the page reloads, verify you see the `jetpack_backup_admin_page_view` event in the console.
- Click the "**See backups in the cloud**" CTA button. Verify the `jetpack_backup_see_backups_cta_click` event fires.
- Click the "**Or view your most recent restore point**" link, verify the `jetpack_backup_view_recent_restore_points_click` event fires.
- Click the "**Learn more**" link. Verify the `jetpack_backup_learn_more_click` event fires.
- Click the "**See all your backups**" link. Verify the `jetpack_backup_see_all_backups_click` event fires.
- Click the "**See your site's activity**" link. Verify the `jetpack_backup_see_site_activity_click` event fires.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202580352248681
